beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -281,40 +259,84 @@ func (s *Server) SubmitAttestationsV2(w http.ResponseWriter, r *http.Request) {
 
 func (s *Server) handleAttestationsElectra(
 	ctx context.Context,
-	data json.RawMessage,
+	r *http.Request,
 ) (attFailures []*server.IndexedVerificationFailure, failedBroadcasts []string, err error) {
-	var sourceAttestations []*structs.SingleAttestation
+	var validAttestations []*eth.SingleAttestation
 	currentEpoch := slots.ToEpoch(s.TimeFetcher.CurrentSlot())
 	if currentEpoch < params.BeaconConfig().ElectraForkEpoch {
 		return nil, nil, errors.Errorf("electra attestations have not been enabled, current epoch %d enabled epoch %d", currentEpoch, params.BeaconConfig().ElectraForkEpoch)
 	}
 
-	if err = json.Unmarshal(data, &sourceAttestations); err != nil {
-		return nil, nil, errors.Wrap(err, "failed to unmarshal attestation")
-	}
+	if httputil.IsRequestSsz(r) {
+		sszLen := (&eth.SingleAttestation{}).SizeSSZ()
+		body, err := readRequestBody(r)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "Could not read request body")
+		}
+		if len(body) < sszLen {
+			return nil, nil, errors.New("No data submitted")
+		}
 
-	if len(sourceAttestations) == 0 {
-		return nil, nil, errors.New("no data submitted")
-	}
+		i := 0
+		for len(body) >= sszLen {
+			var sourceAtt eth.SingleAttestation
+			if err := sourceAtt.UnmarshalSSZ(body[:sszLen]); err != nil {
+				return nil, nil, errors.Wrap(err, "Could not unmarshal ssz attestation")
+			}
+			body = body[sszLen:]
+
+			if _, err = bls.SignatureFromBytes(sourceAtt.Signature); err != nil {
+				attFailures = append(attFailures, &server.IndexedVerificationFailure{
+					Index:   i,
+					Message: "Incorrect attestation signature: " + err.Error(),
+				})
+				i++
+				continue
+			}
+			validAttestations = append(validAttestations, &sourceAtt)
+			i++
```

Comment:
```suggestion
			else {
				validAttestations = append(validAttestations, &sourceAtt)
			}
			i++
```
1 line saved :sunglasses: